ui/components/component-library/popover/popover.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, forwardRef, useImperativeHandle, Ref } from 'react';
+import { createPortal } from 'react-dom';
+import { usePopper } from 'react-popper';
+import classnames from 'classnames';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  DISPLAY,
+  JustifyContent,
+} from '../../../helpers/constants/design-system';
+import Box from '../../ui/box/box';
+import { PopoverPosition, PopoverProps, PopoverRole } from '.';
+
+interface PopoverRef {
+  closePopover: () => void;
+}
+
+export const Popover = forwardRef<PopoverRef, PopoverProps>(
+  (
+    {
+      children,
+      className = '',
+      position = PopoverPosition.Auto,
+      role = PopoverRole.Tooltip,
+      hasArrow = false,
+      matchWidth,
+      preventOverflow = false,
+      offset = [0, 8],
+      flip = false,
+      referenceHidden = true,
+      referenceElement,
+      isOpen,
+      title,
+      isPortal = true,
+      arrowProps,
+      ...props
+    }: PopoverProps,
+    ref: Ref<PopoverRef>,
+  ) => {
+    const [popperElement, setPopperElement] = useState<HTMLElement | null>(
+      null,
+    );
+    const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
+
+    // Define Popper options
+    const { styles, attributes } = usePopper(referenceElement, popperElement, {
+      placement: position,
+      modifiers: [
+        {
+          name: 'preventOverflow',
+          enabled: position === PopoverPosition.Auto ? true : preventOverflow,
+        },
+        {
+          name: 'flip',
+          enabled: position === PopoverPosition.Auto ? true : flip,
+        },
+        {
+          name: 'arrow',
+          enabled: hasArrow,
+          options: {
+            element: arrowElement,
+          },
+        },
+        {
+          name: 'offset',
+          options: {
+            offset,
+          },
+        },
+      ],
+    });
+
+    // Define width to match reference element or auto
+    const contentStyle = {
+      width: matchWidth ? referenceElement?.clientWidth : 'auto',
+    };
+
+    // Forwarding function to close the popover using the ref
+    useImperativeHandle(ref, () => ({
+      closePopover: () => {
+        // Your close popover implementation here
+        console.log('closePopover called');
+      },
+    }));
```

Comment:
Is this still needed? It doesn't look like it does anything other than console log?